core/deployment/src/main/java/io/quarkus/deployment/pkg/PackageConfig.java

```diff
@@ -56,6 +57,37 @@
     @ConfigItem
     public Optional<List<String>> userConfiguredIgnoredEntries;
 
+    /**
+     * List of all the dependencies that have been defined as optional to include into the final package of the application.
+     * Each optional dependency needs to be expressed in the following format:
+     * <p>
+     * groupId:artifactId:classifier:type
+     * <p>
+     * With the classifier and type being optional.
+     * <p>
+     * If the type is missing, the artifact is assumed to be of type {@code jar}.
+     * <p>
+     * This parameter is optional, if absent, no optional dependencies will be included into the final package of
+     * the application.
+     * <p>
+     * For backward compatibility reasons, this parameter is ignored by default and can be enabled by setting the
```

Comment:
Given that this config option is new, could you please clarify what the backward compatibility reasons actually are?